[2024-04-04T11:56:54.506+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=non-requeueable deps ti=<TaskInstance: conrad_coding_test.task_2_spark manual__2024-04-04T11:55:37.452221+00:00 [queued]>
[2024-04-04T11:56:54.514+0000] {taskinstance.py:1979} INFO - Dependencies all met for dep_context=requeueable deps ti=<TaskInstance: conrad_coding_test.task_2_spark manual__2024-04-04T11:55:37.452221+00:00 [queued]>
[2024-04-04T11:56:54.514+0000] {taskinstance.py:2193} INFO - Starting attempt 2 of 2
[2024-04-04T11:56:54.531+0000] {taskinstance.py:2217} INFO - Executing <Task(DockerOperator): task_2_spark> on 2024-04-04 11:55:37.452221+00:00
[2024-04-04T11:56:54.540+0000] {standard_task_runner.py:60} INFO - Started process 139690 to run task
[2024-04-04T11:56:54.551+0000] {standard_task_runner.py:87} INFO - Running: ['airflow', 'tasks', 'run', 'conrad_coding_test', 'task_2_spark', 'manual__2024-04-04T11:55:37.452221+00:00', '--job-id', '17', '--raw', '--subdir', 'DAGS_FOLDER/dag_spark_multi_source_test_docker.py', '--cfg-path', '/tmp/tmptsnnpm8r']
[2024-04-04T11:56:54.556+0000] {standard_task_runner.py:88} INFO - Job 17: Subtask task_2_spark
[2024-04-04T11:56:54.615+0000] {task_command.py:423} INFO - Running <TaskInstance: conrad_coding_test.task_2_spark manual__2024-04-04T11:55:37.452221+00:00 [running]> on host de-zoomcamp.us-central1-a.c.my-zhe-414813.internal
[2024-04-04T11:56:54.685+0000] {taskinstance.py:2513} INFO - Exporting env vars: AIRFLOW_CTX_DAG_OWNER='airflow' AIRFLOW_CTX_DAG_ID='conrad_coding_test' AIRFLOW_CTX_TASK_ID='task_2_spark' AIRFLOW_CTX_EXECUTION_DATE='2024-04-04T11:55:37.452221+00:00' AIRFLOW_CTX_TRY_NUMBER='2' AIRFLOW_CTX_DAG_RUN_ID='manual__2024-04-04T11:55:37.452221+00:00'
[2024-04-04T11:56:54.714+0000] {docker.py:359} INFO - Starting docker container from image sparkhome:latest
[2024-04-04T11:56:55.225+0000] {docker.py:429} INFO - Unsetting extraneous env vars (UTC): 11:56:55
[2024-04-04T11:56:55.227+0000] {docker.py:429} INFO - Finished unsetting extraneous env vars (UTC): 11:56:55
[2024-04-04T11:56:55.233+0000] {docker.py:429} INFO - ++ id -u
[2024-04-04T11:56:55.239+0000] {docker.py:429} INFO - + myuid=0
[2024-04-04T11:56:55.240+0000] {docker.py:429} INFO - ++ id -g
[2024-04-04T11:56:55.243+0000] {docker.py:429} INFO - + mygid=0
[2024-04-04T11:56:55.243+0000] {docker.py:429} INFO - + set +e
[2024-04-04T11:56:55.244+0000] {docker.py:429} INFO - ++ getent passwd 0
[2024-04-04T11:56:55.248+0000] {docker.py:429} INFO - + uidentry=root:x:0:0:root:/root:/bin/bash
[2024-04-04T11:56:55.249+0000] {docker.py:429} INFO - + set -e
+ '[' -z root:x:0:0:root:/root:/bin/bash ']'
+ SPARK_CLASSPATH=':/opt/spark/jars/*'
[2024-04-04T11:56:55.250+0000] {docker.py:429} INFO - + env
[2024-04-04T11:56:55.250+0000] {docker.py:429} INFO - + sed 's/[^=]*=\(.*\)/\1/g'
[2024-04-04T11:56:55.252+0000] {docker.py:429} INFO - + grep SPARK_JAVA_OPT_
[2024-04-04T11:56:55.254+0000] {docker.py:429} INFO - + sort -t_ -k4 -n
[2024-04-04T11:56:55.258+0000] {docker.py:429} INFO - + readarray -t SPARK_EXECUTOR_JAVA_OPTS
[2024-04-04T11:56:55.259+0000] {docker.py:429} INFO - + '[' -n '' ']'
[2024-04-04T11:56:55.259+0000] {docker.py:429} INFO - + '[' -z ']'
[2024-04-04T11:56:55.260+0000] {docker.py:429} INFO - + '[' -z ']'
[2024-04-04T11:56:55.260+0000] {docker.py:429} INFO - + '[' -n '' ']'
[2024-04-04T11:56:55.260+0000] {docker.py:429} INFO - + '[' -z ']'
[2024-04-04T11:56:55.260+0000] {docker.py:429} INFO - + '[' -z ']'
[2024-04-04T11:56:55.261+0000] {docker.py:429} INFO - + '[' -z x ']'
[2024-04-04T11:56:55.261+0000] {docker.py:429} INFO - + SPARK_CLASSPATH='/opt/spark/conf::/opt/spark/jars/*'
[2024-04-04T11:56:55.262+0000] {docker.py:429} INFO - Non-spark-on-k8s command provided, proceeding in pass-through mode...
[2024-04-04T11:56:55.262+0000] {docker.py:429} INFO - + case "$1" in
+ echo 'Non-spark-on-k8s command provided, proceeding in pass-through mode...'
+ CMD=("$@")
+ exec /usr/bin/tini -s -- python /opt/spark/etl_task_2.py
[2024-04-04T11:56:58.620+0000] {docker.py:429} INFO - WARNING: An illegal reflective access operation has occurred
WARNING: Illegal reflective access by org.apache.spark.unsafe.Platform (file:/opt/spark/jars/spark-unsafe_2.12-3.2.1.jar) to constructor java.nio.DirectByteBuffer(long,int)
WARNING: Please consider reporting this to the maintainers of org.apache.spark.unsafe.Platform
WARNING: Use --illegal-access=warn to enable warnings of further illegal reflective access operations
WARNING: All illegal access operations will be denied in a future release
[2024-04-04T11:56:58.621+0000] {docker.py:429} INFO - 
[2024-04-04T11:56:59.299+0000] {docker.py:429} INFO - Using Spark's default log4j profile: org/apache/spark/log4j-defaults.properties
[2024-04-04T11:56:59.300+0000] {docker.py:429} INFO - Setting default log level to "WARN
[2024-04-04T11:56:59.300+0000] {docker.py:429} INFO - ".
[2024-04-04T11:56:59.300+0000] {docker.py:429} INFO - To adjust logging level use sc.setLogLevel(newLevel). For SparkR, use setLogLevel(newLevel).
[2024-04-04T11:56:59.301+0000] {docker.py:429} INFO - 
[2024-04-04T11:56:59.661+0000] {docker.py:429} INFO - 24/04/04 11:56:59 WARN NativeCodeLoader: Unable to load native-hadoop library for your platform... using builtin-java classes where applicable
[2024-04-04T11:57:02.855+0000] {docker.py:429} INFO - Traceback (most recent call last):
  File "/opt/spark/etl_task_2.py", line 17, in <module>
    df_1_users = spark.read.option("header", "true").csv('/mount_data/users.csv')
NameError: name 'spark' is not defined
[2024-04-04T11:57:03.172+0000] {taskinstance.py:2731} ERROR - Task failed with exception
Traceback (most recent call last):
  File "/home/datatalks_jan/anaconda3/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 444, in _execute_task
    result = _execute_callable(context=context, **execute_callable_kwargs)
             ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/datatalks_jan/anaconda3/lib/python3.11/site-packages/airflow/models/taskinstance.py", line 414, in _execute_callable
    return execute_callable(context=context, **execute_callable_kwargs)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/datatalks_jan/anaconda3/lib/python3.11/site-packages/airflow/providers/docker/operators/docker.py", line 502, in execute
    return self._run_image()
           ^^^^^^^^^^^^^^^^^
  File "/home/datatalks_jan/anaconda3/lib/python3.11/site-packages/airflow/providers/docker/operators/docker.py", line 364, in _run_image
    return self._run_image_with_mounts([*self.mounts, tmp_mount], add_tmp_variable=True)
           ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
  File "/home/datatalks_jan/anaconda3/lib/python3.11/site-packages/airflow/providers/docker/operators/docker.py", line 437, in _run_image_with_mounts
    raise DockerContainerFailedException(f"Docker container failed: {result!r}", logs=log_lines)
airflow.providers.docker.exceptions.DockerContainerFailedException: Docker container failed: {'StatusCode': 1}
[2024-04-04T11:57:03.179+0000] {taskinstance.py:1149} INFO - Marking task as FAILED. dag_id=conrad_coding_test, task_id=task_2_spark, execution_date=20240404T115537, start_date=20240404T115654, end_date=20240404T115703
[2024-04-04T11:57:03.201+0000] {standard_task_runner.py:107} ERROR - Failed to execute job 17 for task task_2_spark (Docker container failed: {'StatusCode': 1}; 139690)
[2024-04-04T11:57:03.247+0000] {local_task_job_runner.py:234} INFO - Task exited with return code 1
[2024-04-04T11:57:03.269+0000] {taskinstance.py:3312} INFO - 0 downstream tasks scheduled from follow-on schedule check
